test/contract_tests.js

```diff
@@ -0,0 +1,543 @@
+            await expect(Einstance.connect(ERC20Payer).deposit(reportRoot, tkk_instance.address, ONE_ETHER_FORMAT))
+            Before_Balance = await Einstance.connect(Triager).getBalance(reportRoot, tkk_instance.address);
+            
+            await expect(Einstance.connect(Reporter2).withdraw(reportRoot, tkk_instance.address, HALF_ETHER_FORMAT, salt, reportAddress, merkleProofval))
+            After_Balance = await Einstance.connect(Triager).getBalance(reportRoot, tkk_instance.address);
+            await expect(ethers.utils.formatEther(After_Balance)).to.equal("0.5");
+        })
+
+        it("withdraw(): multiple withdrawals on same report should work", async function () {
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 5, salt, reportAddress, merkleProofval))
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 5, salt, reportAddress, merkleProofval))
+        })
+
+        it("withdraw(): Withdrawing unpaid Report should revert", async function () {
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 5, salt, reportAddress, merkleProofval)).to.be.reverted;
+        })
+
+        it("withdraw(): Should work with Withdrawal amount with '0'", async function () { // Since there's no check for amount 0, Transaction gonna utilize the gas only then
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 0, salt, reportAddress, merkleProofval));
+        })
+
+        it("withdraw(): revert if withdrawing with invalid Payment/Token Address",async function(){
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee, 10, salt, reportAddress, merkleProofval)).to.be.reverted;
+        })
+
+        it("withdraw(): revert if withdrawal amount > report amount", async function(){
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 1000, salt, reportAddress, merkleProofval)).to.be.reverted;
+        })
+
+        it("withdraw(): Anyone can perform withdraw() on the report", async function () {  // but the payments goes to original report owner who provided walletaddress
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, ONE_ETHER_FORMAT, { value: ONE_ETHER_FORMAT })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, HALF_ETHER_FORMAT, salt, reportAddress, merkleProofval))
+        })
```

Comment:
Can we also check that a withdrawal issued from an account other than the reporter results in only the _reporter_ receiving the funds?